### ISSUE TITLE
Shatter combo breaks at max range

### ISSUE THREAD

**Issue by vituscze** [SimCMinMax/WoW-BugTracker#609]
**Game Version:**
> 9.0.1 beta Build 35679

**Description:**
> With sufficient haste, it is possible for all 3 Flurry projectiles to hit before the preceding spell (Frostbolt, Ebonbolt or Glacial Spike) hits. The spell then consumes one of the Winter's Chill stacks, making it impossible to shatter two additional spells cast after Flurry.

**Logs, Images, Videos**
![image](https://user-images.githubusercontent.com/4149104/91895886-3fee7e00-ec98-11ea-9bd1-c2a0c2f66834.png)

**Reproducible:**
> yes

**Howto:**
> stand at max range, cast Icy Veins and Time Warp, cast Ebonbolt and Flurry, after Ebonbolt and Flurry hit, only one stack of Winter's Chill is present; same can be done with Frostbolt (and Glacial Spike) but the haste requirement is higher

**Comment by vituscze** [comment]
Another example on a boss with large hitbox, this time with Frostbolt.
![image](https://user-images.githubusercontent.com/4149104/91898005-682bac00-ec9b-11ea-94ec-a698620b76c6.png)

**Comment by DiggidyDale** [comment]
think this is the same bug as #697

**Comment by vituscze** [comment]
This happened prior to the change that caused #697, but it is related.

**Comment by DiggidyDale** [comment]
Yeah I can see it was happening before but to me at least they seem to have the same cause. Which is projectile timing and the fact only the first flurry applies the buff.
I think that is kind of intended to stop people gaming it to get 3 or more shatters from the brainfreeze proc. Granted they could come up with a better way to stop that.
